sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTest.java

```diff
@@ -212,6 +276,45 @@ public void testAvroIOWriteAndReadSchemaUpgrade() throws Throwable {
     p.run();
   }
 
+  @Test
+  public void testWriteWithDefaultCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz");
+    assertTrue(write.getCodec().toString().equals(CodecFactory.deflateCodec(6).toString()));
+  }
+
+  @Test
+  public void testWriteWithCustomCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz")
+        .withCodec(CodecFactory.snappyCodec());
+    assertTrue(write.getCodec().toString().equals(SNAPPY_CODEC));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testWriteWithSerDeCustomDeflateCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz")
+        .withCodec(CodecFactory.deflateCodec(9));
+
+    AvroIO.Write.Bound<GenericRecord> serdeWrite = SerializableUtils.clone(write);
+
+    assertTrue(serdeWrite.getCodec().toString().equals(CodecFactory.deflateCodec(9).toString()));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testWriteWithSerDeCustomXZCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz")
+        .withCodec(CodecFactory.xzCodec(9));
+
+    AvroIO.Write.Bound<GenericRecord> serdeWrite = SerializableUtils.clone(write);
+
+    assertTrue(serdeWrite.getCodec().toString().equals(CodecFactory.xzCodec(9).toString()));
+  }
```

Comment:
The reason why I compare String is because:

```
scala> import org.apache.avro.file.CodecFactory
import org.apache.avro.file.CodecFactory
scala> CodecFactory.deflateCodec(1).equals(CodecFactory.deflateCodec(1))
res5: Boolean = false
```